doc/userguide/devguide/internals/pipeline/packet-pipeline.rst

```diff
@@ -0,0 +1,62 @@
+===============
+Packet Pipeline
+===============
+
+Main Takeaways:
+
+- What is Suricata's Packet Pipeline
+- Main components
+- How do these work together
+- Different thread models (?)
+
+Introduction
+============
+
+The Packet Pipeline might be considered the core part of Suricata. It is in this
+macro-component that the Packet processing happen (``PacketProcessing``
+threads). The two other main components are **Management threads**
+responsible for *Flow Managing* and *Stats*.
+
+In Suricata, the packet pipeline starts with the Packet Capture.
+
+The pipeline either runs in a single thread (runmode *single* or *workers*) or
+split in 2 stages (*autofp*).
+
+A pipeline is created by Suricata's ``RunMode`` and stored in per thread
```

Comment:
not sure where, but it would be good to mention we use a "run to completion" pipeline. So one packet is processed fully, then the next, then the next, etc. For workers/single this is full "pool to pool", so packet is taken from pool by capture processed and returned to pool. In autofp the pipeline is split. The Capture/decode sub-pipeline run to completion as well, as well as the flowworker/reject/verdict sub pipeline on the worker threads.